### ISSUE TITLE
`unset_binary` function does not work after `optimize!` is called

### ISSUE THREAD

**Issue by dlcole3** [plasmo-dev/Plasmo.jl#87]
I found that calling `unset_binary` does not work after `optimize!` is called when using Juniper as the solver. Below is a MWE that returns the same error. 

```
using Ipopt, Juniper, Plasmo

graph = OptiGraph()

@optinode(graph, nodes[1:2])
@variable(nodes[1], x, Bin)
@variable(nodes[1], 1 <= y)
@variable(nodes[2], 1 <= y)
@linkconstraint(graph, nodes[1][:y] + nodes[2][:y] >= 3)
@constraint(nodes[1], nodes[1][:x] + nodes[1][:y] == 1.5)
@objective(graph, Min, nodes[1][:y] + nodes[2][:y] - nodes[1][:x])

ipopt = optimizer_with_attributes(Ipopt.Optimizer, "print_level"=>0)
optimizer = optimizer_with_attributes(Juniper.Optimizer, "nl_solver"=>ipopt)

set_optimizer(graph, optimizer)
optimize!(graph)
unset_binary(graph[:nodes][1][:x])
```
I initially thought it might be a problem on Juniper's side, but I can run the same problem in JuMP using Juniper and it does not return an error (see below).
```
using JuMP
m = Model()

@variable(m, x, Bin)
@variable(m, y1 >= 1)
@variable(m, y2 >= 1)
@constraint(m, y1 + y2 >= 3)
@constraint(m, y1 + x == 1.5)
@objective(m, Min, y1 + y2 - x)

set_optimizer(m, optimizer)

optimize!(m)
unset_binary(m[:x])
```
I followed the stacktrace, and it seems like it might be because [this line](https://github.com/plasmo-dev/Plasmo.jl/blob/37f80fe96781dc647b3a5e0ed26b9eb0a994bc4a/src/moi_backend_node.jl#L127) calls `delete` on a `MOI.Bridges.LazyBridgeOptimizer` object. In contrast, [this line](https://github.com/plasmo-dev/Plasmo.jl/blob/37f80fe96781dc647b3a5e0ed26b9eb0a994bc4a/src/moi_backend_node.jl#L123) works, but it is calling `delete` on a `MOI.Utilities.CachingOptimizer` object. It looks like the JuMP version also only calls `delete` on the `CachingOptimizer` object.

**Comment by jalving** [comment]
With v0.5.4, you can use a work-around with `reset_optimizer` like following:

```
using Ipopt, Juniper, Plasmo

graph = OptiGraph()

@optinode(graph, nodes[1:2])
@variable(nodes[1], x, Bin)
@variable(nodes[1], 1 <= y)
@variable(nodes[2], 1 <= y)
@linkconstraint(graph, nodes[1][:y] + nodes[2][:y] >= 3)
@constraint(nodes[1], nodes[1][:x] + nodes[1][:y] == 1.5)
@objective(graph, Min, nodes[1][:y] + nodes[2][:y] - nodes[1][:x])

ipopt = optimizer_with_attributes(Ipopt.Optimizer, "print_level"=>0)
optimizer = optimizer_with_attributes(Juniper.Optimizer, "nl_solver"=>ipopt)

set_optimizer(graph, optimizer)
optimize!(graph)
reset_optimizer(graph)
unset_binary(graph[:nodes][1][:x])
```
Juniper does not seem to support deleting constraints, so it falls back to the caching optimizer. We will need to change how we manage optimizers to fix this correctly. Mainly, we should call `reset_optimizer` when we detect unsupported changes. 

Your first snippet does work with HiGHS which supports deleting constraints.

**Comment by jalving** [comment]
This seems to be fixed in v0.6 @dlcole3.  Please re-open if you run into this issue again.
